be/src/storage/rowset/segment_v2/column_reader.cpp

```diff
@@ -1101,7 +1101,17 @@ Status DefaultValueColumnIterator::next_batch(size_t* n, vectorized::Column* dst
         _current_rowid += *n;
         DCHECK(ok) << "cannot append null to non-nullable column";
     } else {
-        dst->append_value_multiple_times(_mem_value, *n);
+        if (_type_info->type() == OLAP_FIELD_TYPE_OBJECT || _type_info->type() == OLAP_FIELD_TYPE_HLL ||
+            _type_info->type() == OLAP_FIELD_TYPE_PERCENTILE) {
+            std::vector<Slice> slices;
+            slices.reserve(*n);
+            for (size_t i = 0; i < *n; i++) {
+                slices.emplace_back(*reinterpret_cast<const Slice*>(_mem_value));
```

Comment:
I have a question. Why not deserialize it as _mem_value before and call 'append_value_multiple_times' here.

Reply:
append_strings and append_value_multiple_times on the same level, can instead it comprehensibly.
https://github.com/StarRocks/starrocks/blob/2c57789c16e0e3be00e69527969ba3eef8b8af70/be/src/column/object_column.cpp#L90-L111

this level can do not know about bitmap value implement according to least knowledge principle.

https://github.com/StarRocks/starrocks/blob/2c57789c16e0e3be00e69527969ba3eef8b8af70/be/src/storage/rowset/segment_v2/column_reader.cpp#L1051-L1058

https://github.com/StarRocks/starrocks/blob/5fa55b8199f3b50fbb06ad686f979ec9d87da37d/be/src/util/bitmap_value.h#L991